tinygrad/runtime/ops_hip.py

```diff
@@ -100,6 +104,7 @@ def __init__(self, name:str, prg:str, binary=False):
       self.modules.append(hip.hipModuleLoadData(prg))
       self.prgs.append(hip.hipModuleGetFunction(self.modules[-1], name))
 
+  cache_compiled = lambda _: _ if getenv("HIP_CPU") else cache_compiled # noqa: E731
```

Comment:
Why does the cache break with CPU?

Reply:
because hiprtcCreateProgram returns the CDLL (and write_bytes expects byte-like objects)